server/src/test/java/org/elasticsearch/indices/cluster/IndicesClusterStateServiceRandomUpdatesTests.java

```diff
@@ -99,6 +107,44 @@ public void tearDown() throws Exception {
         terminate(threadPool);
     }
 
+    public void testRepository() {
+        ClusterState state = ClusterStateCreationUtils.state(3, new String[]{"test-index"}, randomIntBetween(1, 5));
+        state = cluster.putRepository(
+            state,
+            new PutRepositoryRequest("test-repo").type("fs").verify(false)
+                .settings(
+                    "{\"location\": \"" + Environment.PATH_REPO_SETTING.get(ClusterStateChanges.SETTINGS).get(0) + "\"}",
+                    XContentType.JSON
+                )
+        );
+        state = cluster.createSnapshot(state, new CreateSnapshotRequest("test-repo", "test-snap"));
+        assertNotNull(state.custom(SnapshotsInProgress.TYPE));
+        SnapshotsInProgress.Entry firstEntry = ((SnapshotsInProgress) state.custom(SnapshotsInProgress.TYPE)).entries().get(0);
+        assertEquals("test-snap", firstEntry.snapshot().getSnapshotId().getName());
+        assertEquals(SnapshotsInProgress.State.INIT, firstEntry.state());
+        state = cluster.beginSnapshot(state, firstEntry, false);
+        SnapshotsInProgress.Entry secondEntry = ((SnapshotsInProgress) state.custom(SnapshotsInProgress.TYPE)).entries().get(0);
+        assertEquals(Strings.toString(state, true, true), SnapshotsInProgress.State.STARTED, secondEntry.state());
+        final ClusterState finalState = state;
+        RuntimeException ex =
+            expectThrows(RuntimeException.class, () -> cluster.deleteIndices(finalState, new DeleteIndexRequest("test-index")));
+        assertEquals(
+            "Cannot delete indices that are being snapshotted: [[test-index]]. " +
+                "Try again after snapshot finishes or cancel the currently running snapshot.",
+            ex.getCause().getMessage()
+        );
+        state = cluster.deleteSnapshot(state, new DeleteSnapshotRequest("test-repo", "test-snap"));
+        assertNotNull(state.custom(SnapshotsInProgress.TYPE));
+        // TODO: This should be 1, but we're currently leaving one aborted entry behind
```

Comment:
what would it take to fix this? What other tests could we possibly add here? Move this tests (and other new ones) to their own test class?